sandbox/cgroup/cgroup.go

```diff
@@ -109,9 +109,67 @@ func Version() (int, error) {
 	return probeVersion, probeErr
 }
 
+// GroupMatcher attempts to match the cgroup entry
+type GroupMatcher interface {
+	String() string
+	// Match returns true when given tuple of hierarchy-ID and controllers is a match
+	Match(id, maybeControllers string) bool
+}
+
+type unified struct{}
+
+func (u *unified) Match(id, maybeControllers string) bool {
+	return id == "0" && maybeControllers == ""
+}
+func (u *unified) String() string { return "unified hierarchy" }
+
+// MatchUnifiedHierarchy provides matches for unified cgroup hierarchies
+func MatchUnifiedHierarchy() GroupMatcher {
+	return &unified{}
+}
+
+type v1NamedHierarchy struct {
+	name string
+}
+
+func (n *v1NamedHierarchy) Match(_, maybeControllers string) bool {
+	if !strings.HasPrefix(maybeControllers, "name=") {
+		return false
+	}
+	name := strings.TrimPrefix(maybeControllers, "name=")
+	return name == n.name
+}
+
+func (n *v1NamedHierarchy) String() string { return fmt.Sprintf("named hierarchy %q", n.name) }
+
+// MatchV1NamedHierarchy provides a matcher for a given named v1 hierarchy
+func MatchV1NamedHierarchy(hierarchyName string) GroupMatcher {
+	return &v1NamedHierarchy{name: hierarchyName}
+}
+
+type v1Controller struct {
+	controller string
+}
+
+func (n *v1Controller) Match(_, maybeControllers string) bool {
+	controllerList := strings.Split(maybeControllers, ",")
+	return strutil.ListContains(controllerList, n.controller)
+}
+
+func (n *v1Controller) String() string { return fmt.Sprintf("controller %q", n.controller) }
+
+// MatchV1Controller provides a matches for a given v1 controller
+func MatchV1Controller(controller string) GroupMatcher {
+	return &v1Controller{controller: controller}
+}
+
 // ProcGroup finds the path of a given cgroup controller for provided process
 // id.
-func ProcGroup(pid int, controller string) (string, error) {
+func ProcGroup(pid int, matcher GroupMatcher) (string, error) {
```

Comment:
Silly question - is this is the only function that takes the new GroupMatcher? Or will this be extended? If its the only user of this maybe we can leave the ProcGroup() API identical (i.e. it takes a string) and we just create the right matcher internally in ProcGroup() ?